db/python/layers/billing.py

```diff
@@ -424,3 +509,332 @@ async def get_total_cost(
+                day <= TIMESTAMP(@last_day)
+                {invoice_month_filter}
+                GROUP BY
+                field,
+                cost_category
+            )
+            WHERE
+            cost > 0.1
+        ) month
+        {daily_cost_join}
+        ORDER BY field ASC, daily_cost DESC, monthly_cost DESC;
+        """
+
+        query_params = [
+            bigquery.ScalarQueryParameter('start_day', 'STRING', start_day),
+            bigquery.ScalarQueryParameter('last_day', 'STRING', last_day),
+        ]
+
+        if is_current_month:
+            query_params.append(
+                bigquery.ScalarQueryParameter(
+                    'last_loaded_day', 'STRING', last_loaded_day
+                ),
+            )
+
+        if has_valid_invoice_month:
+            query_params.append(
+                bigquery.ScalarQueryParameter('invoice_month', 'STRING', invoice_month)
+            )
+
+        return is_current_month, last_loaded_day, list(
+            self._connection.connection.query(
+                _query,
+                job_config=bigquery.QueryJobConfig(query_parameters=query_params),
+            ).result()
+        )
+
+    async def get_running_cost(
+        self, field: BillingColumn, invoice_month: str | None = None
+    ) -> list[BillingCostBudgetRecord]:
+        """
+        Get currently running cost of selected field
+        """
+
+        # accept only Topic, Dataset or Project at this stage
+        if field not in (
+            BillingColumn.TOPIC,
+            BillingColumn.PROJECT,
+            BillingColumn.DATASET,
+        ):
+            raise ValueError('Invalid field')
+
+        (
+            is_current_month,
+            last_loaded_day,
+            query_job_result
+        ) = await self.execute_running_cost_query(
+            field, invoice_month
+        )
+        if not query_job_result:
+            # return empty list
+            return []
+
+        # prepare data
+        results = []
+
+        # reformat last_loaded_day if present
+        if last_loaded_day:
+            last_loaded_day = datetime.strptime(last_loaded_day, '%Y-%m-%d %H:%M:%S+00:00')
+            last_loaded_day = last_loaded_day.strftime('%b %d')
+
+        total_monthly: dict[str, Counter[str]] = {
+            'C': Counter(),
+            'S': Counter(),
+        }
+        total_daily: dict[str, Counter[str]] = {
+            'C': Counter(),
+            'S': Counter(),
+        }
+        field_details: dict[str, list[Any]] = {}  # detail category cost for each field
+        total_monthly_category: Counter[str] = Counter()
+        total_daily_category: Counter[str] = Counter()
+        for row in query_job_result:
+            if row.field not in field_details:
+                field_details[row.field] = []
+
+            field_details[row.field].append(
+                {
+                    'cost_group': 'S' if row.cost_category == 'Cloud Storage' else 'C',
```

Comment:
Can you please factor this out into a method that takes the cost_category and returns `'C' | 'S'`? Saves this logic from being buried in here, it's used a few times in this method, and I'm a bit wary of google cloud constant string value.

Also means the below can be simplified.